管理者用ダッシュボードの試作を追加

shimizu-kiichi opened the pull request:
# 概要
<!-- このプルリクエストの目的や概要を簡潔に記述してください -->
管理者用ダッシュボードを試作を追加しました。
# 変更点
<!-- 具体的な変更内容をリスト形式で記述してください -->

# 関連タスク
<!-- 関連するイシューがあれば、番号を記述してください (例: #123) -->

# 確認事項
- [ ] コーディング規約を遵守している
- [x] 自身のコードをセルフレビューした
- [x] 動作確認を実施し、想定通りの挙動を確認した
- [ ] 関連するドキュメント（README.mdなど）を更新した

# 備考
<!-- その他、レビュワーに伝えたいことがあれば記述してください -->
ダッシュボードでstatus列の操作・登録情報の削除機能を実装予定ですが、今のところ未対応です。

<!-- for GitHub Copilot review rule-->

<!-- I want to review in Japanese. -->

shimizu-kiichi committed:
add(dashboard): 管理者用ダッシュボード(Webアプリ)を追加

# スプレッドシートの文字列を取得して、Webアプリ上に表示する

最終的に、管理者用ダッシュボードでスプレッドシートを管理する。そのため、スプレッドシートのセルの文字列をGASのWebアプリで表示できるか確認する。

- `SpreadsheetApp`を使用して対象のシートにアクセス
- `getRange()`と`getValue()`を使って特定のセルの文字列を取得
- `doGet()`でテンプレートHTMLにデータを埋め込み、Webアプリ上に表示

shimizu-kiichi committed:
add(dashboard): スプレッドシート全体の情報をWebアプリに表示

# スプレッドシート全体の情報をWebアプリに表示する

code.gsでスプレッドシートの特定のシート全体の文字列を取得して、Webアプリで表示できるようにする。

- `getDataRange`と`getValues()`を使ってシート全体の文字列を取得
- `doGet()`でテンプレートHTMLにデータを埋め込み、Webアプリ上に表示

shimizu-kiichi committed:
add(dashboard): 特定の列を非表示にする

# Webアプリに表示するテーブルから、指定した列を除外する機能を追加

- `doGet()`で非表示対象の列番号をHTMLテンプレートに渡す
- `HTML`テンプレート側で列番号を判定し、対象の `<th>` (ヘッダー) と `<td>` (セル) を描画しないよう条件分岐を追加

shimizu-kiichi committed:
add(dashboard): 画像の表示に対応

# スプレッドシートに記録されたGoogle DriveのidからURLを作成し、Webアプリで画像を表示させる

- `doGet()`でidをHTMLテンプレートに渡す
- "https://lh3.googleusercontent.com/d/"の後にidを付け加えたURLを作成
- `img`タグで生成したURLを使い、画像を表示

shimizu-kiichi committed:
add(dashboard): 画像の表示に対応

# スプレッドシートに記録されたGoogle DriveのidからURLを作成し、Webアプリで画像を表示させる

- `doGet()`でidをHTMLテンプレートに渡す
- "https://lh3.googleusercontent.com/d/"の後にidを付け加えたURLを作成
- `img`タグで生成したURLを使い、画像を表示

KazumasaFUJIWARA reviewed (commented):
サンプルページが見えないので、処理できません。
どこで見れますか？

KazumasaFUJIWARA committed:
add: demo-page

KazumasaFUJIWARA committed:
add(dashboard): Enhance demo page with new button layout and email tooltip

- Introduced a header container with buttons for archiving and action groups (complete, discard, delete).
- Added styling for buttons with hover effects for better user interaction.
- Implemented tooltips displaying email addresses on hover for specific table cells.

shimizu-kiichi committed:
refactor(dashboard): copilotによる修正を反映
add(dashboard): index-demo.htmlの一部をindex.htmlに反映

shimizu-kiichi committed:
refactor(dashboard): copilotによる修正を反映
add(dashboard): index-demo.htmlの一部をindex.htmlに反映
